sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/RuleFinder.cs

```diff
@@ -78,28 +85,25 @@ public static IEnumerable<Type> GetUtilityAnalyzerTypes(AnalyzerLanguage languag
                 .Where(type => GetTargetLanguages(type).IsAlso(language));
         }
 
-        public static AnalyzerLanguage GetTargetLanguages(Type analyzerType)
+        private static AnalyzerLanguage GetTargetLanguages(MemberInfo analyzerType)
         {
-            var attribute = analyzerType.GetCustomAttributes<DiagnosticAnalyzerAttribute>().FirstOrDefault();
-            if (attribute == null)
-            {
-                return null;
-            }
+            var languages = GetLanguages(analyzerType);
 
-            var language = AnalyzerLanguage.None;
-            foreach (var lang in attribute.Languages)
+            return languages.Aggregate(AnalyzerLanguage.None, (current, lang) => lang switch
             {
-                if (lang == LanguageNames.CSharp)
-                {
-                    language = language.AddLanguage(AnalyzerLanguage.CSharp);
-                }
-                else if (lang == LanguageNames.VisualBasic)
-                {
-                    language = language.AddLanguage(AnalyzerLanguage.VisualBasic);
-                }
-            }
+                LanguageNames.CSharp => current.AddLanguage(AnalyzerLanguage.CSharp),
+                LanguageNames.VisualBasic => current.AddLanguage(AnalyzerLanguage.VisualBasic),
+                _ => current
+            });
+        }
+
+        private static IEnumerable<string> GetLanguages(MemberInfo analyzerType)
+        {
+            var diagnosticAttribute = analyzerType.GetCustomAttributes<DiagnosticAnalyzerAttribute>().FirstOrDefault();
 
-            return language;
+            return diagnosticAttribute == null
+                ? analyzerType.GetCustomAttributes<RuleAttribute>().FirstOrDefault()?.Languages
```

Comment:
do you want to throw here? in case the language is not provided? and add tests for this case?

it should be caught by unit tests